phabricatoremails/template.py

```diff
@@ -0,0 +1,237 @@
+
+
+def is_code(context):
+    return isinstance(context, CodeContext)
+
+
+def is_has_comments(event):
+    return event.main_comment or event.inline_comments
+
+
+def is_has_comments_secure(event):
+    return event.comment_count > 0
+
+
+def is_reviewer_accepted(reviewer: Reviewer):
+    return not reviewer.is_actionable and reviewer.status == ReviewerStatus.ACCEPTED
+
+
+def is_reviewer_blocking(reviewer: Reviewer):
+    return reviewer.status == ReviewerStatus.BLOCKING
+
+
+def is_reviewer_soft_needed(reviewer: Reviewer):
+    # The revision needs a review from _someone_, and this person _is_ a reviewer, but
+    # it's not them specifically who needs to perform the review
+    return reviewer.is_actionable and reviewer.status == ReviewerStatus.UNREVIEWED
+
+
+def is_reviewer_needed(reviewer: Reviewer):
+    # This specific reviewer needs to accept this revision before it can be landed
+    return reviewer.is_actionable and reviewer.status in [
+        ReviewerStatus.REQUESTED_CHANGES,
+        ReviewerStatus.BLOCKING,
+    ]
+
+
+def reviewer_actionable_class(reviewer: Reviewer):
+    return "actionable" if reviewer.is_actionable else "non-actionable"
+
+
+def date(utc_value: datetime, receiver_timezone: timezone):
+    return utc_value.astimezone(receiver_timezone).strftime("%b %-d %-I:%M%p:")
+
+
+def comment_summary(body):
+    if not body.main_comment and not body.inline_comments:
+        return ""
+    elif (body.main_comment and not body.inline_comments) or (
+        not body.main_comment and len(body.inline_comments) == 1
+    ):
+        return " and submitted a comment"
+    else:
+        return " and submitted comments"
+
+
+def secure_comment_summary(body):
+    if body.comment_count == 0:
+        return ""
+    elif body.comment_count == 1:
+        return " and submitted a comment"
+    else:
+        return " and submitted comments"
+
+
+def diff_class(diff_type: DiffLineType):
+    if diff_type == DiffLineType.ADDED:
+        return "added-line"
+    elif diff_type == DiffLineType.REMOVED:
+        return "removed-line"
+    elif diff_type == DiffLineType.NO_CHANGE:
+        return "no-change-line"
+
+
+def reviewer_status_icon(status: ReviewerStatus):
+    if status == ReviewerStatus.ACCEPTED:
+        return "&#x2714;"
+    elif status == ReviewerStatus.REQUESTED_CHANGES:
+        return "&#x2716;"
+    else:
+        return ""
+
+
+def existence_change_class(reviewer_change: ExistenceChange):
+    if reviewer_change == ExistenceChange.NO_CHANGE:
+        return ""
+    elif reviewer_change == ExistenceChange.ADDED:
+        return "added"
+    elif reviewer_change == ExistenceChange.REMOVED:
+        return "removed"
+
+
+def existence_change_label(reviewer_change: ExistenceChange):
+    if reviewer_change == ExistenceChange.NO_CHANGE:
+        return ""
+    elif reviewer_change == ExistenceChange.ADDED:
+        return "added"
+    elif reviewer_change == ExistenceChange.REMOVED:
+        return "removed"
```

Comment:
Repeated code here. This, and other methods in this module can be refactored differently to reduce code repetition. At the very least you can put these mappings in a dictionary, but these two methods are literally identical in their content.